design-documents/service-isolation.md

```diff
@@ -239,6 +239,9 @@ General principles to follow for service isolation:
 * All service operations MUST BE stateless
 * There MUST BE NO data dependencies between services.
 * Command & query responsibility segregation – storefront APIs for data immutable in storefront (catalog) should be optimized for data retrieval
+* Each request to service MUST contain correlation identifier to allow identify requests related to the same flow during debugging
+* Service mesh MUST BE used for communication between services: routing, load balancing, circuit breaking, statistics
```

Comment:
I wouldn't put it as a principle. It is a solution. It would be good if it goes to a separate document about services communication.